src/components/IPSettingsForm.js

```diff
@@ -36,94 +38,184 @@ import { useNetworkDispatch, updateConnection } from '../NetworkContext';
+ *
+ * Basically, removing duplicated entries and those without IP
+ *
+ * @param {Array<module:model~AddressConfig>} addresses - addresses collection to sanitize
+ * @return {Array<module:model~AddressConfig>}
+ */
 const sanitize = (addresses) => {
-    return addresses.filter((addr, index, collection) => {
-    // Reject addresses without IP
+    return addresses.filter((addr, idx, collection) => {
+        // Reject addresses without IP
         if (addr.local === undefined || addr.local.trim() === "") return false;
 
         // If duplicated (same address, same label), keep only one
-        const idx = collection.findIndex((item) => item.local === addr.local && item.label === addr.label);
-        return idx === index;
+        const firstIdx = collection.findIndex((item) => item.local === addr.local && item.label === addr.label);
+        return idx === firstIdx;
     });
 };
 
+/**
+ * Checks if there is an invalid IP in given addresses collection
+ *
+ * @param {Array<module:model~AddressConfig>} addresses - addresses collection to check
+ * @return {boolean} true if an invalid IP is found; false otherwise
+ */
 const findInvalidIP = addresses => addresses.find((addr) => !isValidIP(addr.local));
 
+/**
+ * Checks if there is a repeated label in given addresses collection
+ *
+ * @param {Array<module:model~AddressConfig>} addresses - addresses collection to check
+ * @return {boolean} true if a label is used more than once; false otherwise
+ */
 const findRepeatedLabel = (addresses) => {
     return addresses.find((addr, idx, collection) => {
         const firstIdx = collection.findIndex((item) => item.label === addr.label);
         return idx !== firstIdx;
     });
 };
 
-const IPSettingsForm = ({ ipVersion = 'ipv4', connection, isOpen, onClose }) => {
+/**
+ * Form to configure the IP settings
+ *
+ * @param {Object} props - form props
+ * @param {object} props.connection - the connection object
+ * @param {string} [props.ipVersion=ipv4] - the IP version
+ * @param {boolean} [props.isOpen=false] - whether the form is displayed or not
+ * @param {function} [props.onClose] - callback function to be called when the form is closed
+ */
+const IPSettingsForm = ({ connection, ipVersion = 'ipv4', isOpen, onClose }) => {
     const dispatch = useNetworkDispatch();
     const settings = connection[ipVersion];
     const [bootProto, setBootProto] = useState(settings.bootProto);
     const [addresses, setAddresses] = useState(settings.addresses);
+    const [addressRequired, setAddressRequired] = useState(settings.bootProto === bootProtocol.STATIC);
     const [errorMessages, setErrorMessages] = useState([]);
     const [isApplying, setIsApplying] = useState(false);
 
-    const handleSubmit = () => {
-        setErrorMessages([]);
-        setIsApplying(true);
+    /**
+     * Performs an update of the internal addresses state
+     *
+     * When the "Static" boot protocol is selected, it ensures that there is at least one {@link
```

Comment:
The thing is that having a 'static' config without IPADDRESS is allowed and even recommended instead of using 'none' as the bootProto when it is not a bond slave.

https://github.com/openSUSE/wicked/blob/ab01fdf1eb1b17ce74bb41dd0bc8186e2aaa4880/man/ifcfg.5.in#L154

Reply:
Ups, I didn't know that at the time to address the issue #27.

Reply:
That is why I said it is fine, I think that even in YaST it is mandatory in the user interface if configured, but was not the case of AutoYaST and a valid use case just to take/have in mind.